### PR TITLE
Split Off Arm64 Perf Pipeline

### PR DESCRIPTION
**Pull request by Tohron**
It seems there are significantly differing loads between perflab arm64 machines and other perflab machines, thus, the yaml is being split in two so that there can be two separate pipelines.

Since perf_slow was merged separately, this PR now removes those entries from perf.yml.

### CODE REVIEW

File: eng/pipelines/coreclr/perf_arm64.yml [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        extraStepsTemplate: /eng/pipelines/common/upload-artifact-step.yml
+        extraStepsParameters:
+          rootFolder: '$(Build.SourcesDirectory)/artifacts/'
+          includeRootFolder: true
+          displayName: Browser Wasm Artifacts
+          artifactName: BrowserWasm
+          archiveType: zip
+          archiveExtension: .zip
+
+  # build mono for AOT
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/perf_arm64.yml:133]: I don't think that we need this step in the ARM64 yml

File: eng/pipelines/coreclr/perf_arm64.yml [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+        extraStepsParameters:
+          rootFolder: '$(Build.SourcesDirectory)/artifacts/'
+          includeRootFolder: true
+          displayName: AOT Mono Artifacts
+          artifactName: LinuxMonoAOTarm64
+          archiveExtension: '.tar.gz'
+          archiveType: tar
+          tarCompression: gz
+
+  # build mono Android scenarios
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/perf_arm64.yml:176]: Same here, this should stay in the regular runs

File: eng/pipelines/coreclr/perf_arm64.yml [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    parameters:
+      jobTemplate: /eng/pipelines/common/build-coreclr-and-libraries-job.yml
+      buildConfig: release
+      platforms:
+      - Linux_x64
+      - windows_x64
+      - windows_x86
+      - Linux_arm64
+      - Linux_musl_x64
+      - windows_arm64
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/perf_arm64.yml:108]: We should get rid of all non-Arm64 builds here.

**Review by DrewScoggins — CHANGES_REQUESTED** [review]

File: eng/pipelines/coreclr/perf_arm64.yml (hunk 2 above)
Comment (Tohron) [reply eng/pipelines/coreclr/perf_arm64.yml:176]: The artifact name
artifactName: AndroidMonoarm64
seems to indicate it involves Arm64

**Comment by DrewScoggins** [comment]
We are also missing the Windows Arm64 runs that are on the PerfSurfs, and the Mono AOT Arm64 run. Can you please include those in the new yaml file, and remove from the old one.

File: eng/pipelines/coreclr/perf_arm64.yml (hunk 2 above)
Comment (DrewScoggins) [reply eng/pipelines/coreclr/perf_arm64.yml:149]: These are the Andoid SoD tests that @billwert added. They take only about 5 minutes to run, so they should not be on the long poll that is Arm64.

Comment (billwert) [reply eng/pipelines/coreclr/perf_arm64.yml]: Correct. Keep in mind this is about balancing queues, not just splitting architectures.

File: eng/pipelines/coreclr/perf_arm64.yml [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+          archiveExtension: .zip
+
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), notin(variables['Build.Reason'], 'Schedule')) }}:
+
+  # build coreclr and libraries
+  - template: /eng/pipelines/common/platform-matrix.yml
+    parameters:
+      jobTemplate: /eng/pipelines/common/build-coreclr-and-libraries-job.yml
+      buildConfig: release
+      platforms:
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/perf_arm64.yml:102]: This also needs to include windows_arm64 as a platform we build for.

File: eng/pipelines/coreclr/perf_arm64.yml [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        extraStepsParameters:
+          rootFolder: '$(Build.SourcesDirectory)/artifacts/'
+          includeRootFolder: true
+          displayName: AOT Mono Artifacts
+          artifactName: LinuxMonoAOTarm64
+          archiveExtension: '.tar.gz'
+          archiveType: tar
+          tarCompression: gz
+
+  # build mono
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/perf_arm64.yml:149]: I think we can get rid of this as well.

**Review by DrewScoggins — CHANGES_REQUESTED** [review]
You need to also include the mono AOT runs in the perf_arm64.yml file.

**Comment by DrewScoggins** [comment]
I think that you should also not rename the perf.yml file, and instead make the perf_x64.yml file a new thing. That way we won't have runs not launching while we are reconfiguring our pipelines. Once we have everything setup the way we like we can go in and delete the perf.yml in another commit.

**Comment by DrewScoggins** [comment]
Can you do a private run of this through AzDO, to ensure that we are seeing the right jobs get cut out, and everything else is parsing correctly?

**Review by billwert — APPROVED** [review]

**Comment by Tohron** [comment]
Need a new PR since Drew is on vacation

**Pull request closed without merging**